internal/config/config.go

```diff
@@ -136,31 +136,91 @@ func initConfig() {
 	} else {
 		viper.SetDefault("LogFormater", "text")
 
-		viper.SetDefault("KAFKA_BOOTSTRAP_SERVERS", "localhost:29092")
-		viper.SetDefault("UPLOAD_TOPIC", "hccm.ros.events")
-		viper.SetDefault("RECOMMENDATION_TOPIC", "rosocp.kruize.recommendations")
-		viper.SetDefault("SOURCES_EVENT_TOPIC", "platform.sources.event-stream")
-
-		// default DB Config
-		viper.SetDefault("DBName", "postgres")
-		viper.SetDefault("DBUser", "postgres")
-		viper.SetDefault("DBPassword", "postgres")
-		viper.SetDefault("DBHost", "localhost")
-		viper.SetDefault("DBPort", "15432")
-		viper.SetDefault("DBssl", "disable")
+		// Kafka Config - read from environment variables first, fallback to defaults
+		kafkaBootstrapServers := os.Getenv("KAFKA_BOOTSTRAP_SERVERS")
+		if kafkaBootstrapServers == "" {
+			kafkaBootstrapServers = "localhost:29092"
+		}
+		uploadTopic := os.Getenv("UPLOAD_TOPIC")
+		if uploadTopic == "" {
+			uploadTopic = "hccm.ros.events"
+		}
+		recommendationTopic := os.Getenv("RECOMMENDATION_TOPIC")
+		if recommendationTopic == "" {
+			recommendationTopic = "rosocp.kruize.recommendations"
+		}
+		sourcesEventTopic := os.Getenv("SOURCES_EVENT_TOPIC")
+		if sourcesEventTopic == "" {
+			sourcesEventTopic = "platform.sources.event-stream"
+		}
+		
+		viper.SetDefault("KAFKA_BOOTSTRAP_SERVERS", kafkaBootstrapServers)
+		viper.SetDefault("UPLOAD_TOPIC", uploadTopic)
+		viper.SetDefault("RECOMMENDATION_TOPIC", recommendationTopic)
+		viper.SetDefault("SOURCES_EVENT_TOPIC", sourcesEventTopic)
+
+		// DB Config - read from environment variables first, fallback to defaults
+		dbHost := os.Getenv("DB_HOST")
+		if dbHost == "" {
+			dbHost = "localhost"
+		}
+		dbPort := os.Getenv("DB_PORT")
+		if dbPort == "" {
+			dbPort = "15432"
+		}
+		dbName := os.Getenv("DB_NAME")
+		if dbName == "" {
+			dbName = "postgres"
+		}
+		dbUser := os.Getenv("DB_USER")
+		if dbUser == "" {
+			dbUser = "postgres"
+		}
+		dbPassword := os.Getenv("DB_PASSWORD")
+		if dbPassword == "" {
+			dbPassword = "postgres"
+		}
+		dbSSL := os.Getenv("DB_SSL")
+		if dbSSL == "" {
+			dbSSL = "disable"
+		}
+		
+		viper.SetDefault("DBName", dbName)
+		viper.SetDefault("DBUser", dbUser)
+		viper.SetDefault("DBPassword", dbPassword)
+		viper.SetDefault("DBHost", dbHost)
+		viper.SetDefault("DBPort", dbPort)
+		viper.SetDefault("DBssl", dbSSL)
```

Comment:
I wonder if we could abstract this flow somehow:
* Retrieve the os env value from a specific env name. If not available default to a specific value (for backwards compatibility)
* Set as default for a given config value for a given config name

It looks like this flow is repeated quite a lot. WDYT?

```
		dbName := os.Getenv("DB_NAME")
		if dbName == "" {
			dbName = "postgres"
		}
		viper.SetDefault("DBName", dbName)
```

to

```
viper.SetDefault("DBName", getValueFromEnvOrDefault("DB_NAME","postgres"))
...
...
}

func getValueFromEnvOrDefault(envKey, default string) string {
		value := os.Getenv(envKey)
		if value == "" {
			return default
		}
        return value
}

Reply:
That's good idea!

Reply:
yeah, it feels overkill for one function: I'd keep one of the functions, probably the first one because we're not parsing the data and we don't care if it has weird characters since we're not processing them.